Roll-out spack-stack-1.5.0

## Description

This PR captures the roll-out of spack-stack-1.5.0.

While we are at it, we should check and update thirdparty, install, and source-cache locations on preconfigured sites for spack-stack-1.5.0.

For example, gaea-c5 and hercules are following the new directory structure for sites that share a filesystem with another site, but their peers gaea-c4 and orion don't.

**Important:** check external packages for sites and remove any older, external `bison` to avoid duplicate packages

Latest release documentation: https://spack-stack.readthedocs.io/en/release-1.5.0/

## Sites

Basic tests are: build jedi-bundle and run ctests **AND/OR** build ufs-weather-model and run regression tests **AND/OR**build ufs-srw-app and run tests (please denote in the list below). If someone wants to test an installation from a different installer with another set of tests, that is more than welcome (just note the github username when you add the test).

- [x] Orion (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location 
    - [x] Installed in: `/work/noaa/epic/role-epic/spack-stack/orion/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers: jedi-bundle with GNU, only two ctests failed out of 1900
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/770
    - [x] (re-)installed with ip@4.3.0
```
The following tests FAILED:
	405 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
	1627 - test_soca_sqrtvertloc (Failed)
```

- [x] Hercules (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in: ``/work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.5.0/envs/unified-env``
    - [x] basic tests performed for all supported compilers (jedi-bundle)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/770
    - [x] (re-)installed with ip@4.3.0
```
The following tests FAILED:
	405 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
	1532 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
	1564 - fv3jedi_test_tier1_diffstates_gfs (Failed)
	1627 - test_soca_sqrtvertloc (Failed)
	1633 - test_soca_dirac_soca_mask (Failed)
	1642 - test_soca_3dvar_godas (Failed)
```

- [x] Discover (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/gpfsm/dswdev/jcsda/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed (jedi-bundle, gnu)
    - [x] site/config and documentation updated:https://github.com/JCSDA/spack-stack/pull/775
    - [x] (re-)installed with ip@4.3.0
```
The following tests FAILED:
        408 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        1541 - fv3jedi_test_tier1_dirac_geos_gsi_global (Failed)
        1542 - fv3jedi_test_tier1_3dvar_geos_sondes (Failed)
        1543 - fv3jedi_test_tier1_3dvar_geos_ozone (Failed)
        1544 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
        1545 - fv3jedi_test_tier1_3dvar_gfs_sondes (Failed)
        1627 - test_soca_setcorscales (Failed)
        1629 - test_soca_parameters_bump_cor_nicas_scales (Failed)
        1632 - test_soca_enspert (Failed)
        1640 - test_soca_sqrtvertloc (Failed)
        1642 - test_soca_dirac_soca_cor_nicas_scales (Failed)
        1644 - test_soca_dirac_socahyb_cov (Failed)
        1646 - test_soca_dirac_soca_mask (Failed)
        1647 - test_soca_dirac_soca_nomask (Failed)
        1655 - test_soca_3dvar_godas (Failed)
```

- [x] Casper (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/glade/work/epicufsrt/contrib/spack-stack/casper/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers: jedi-bundle with Intel, all ctests pass
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/787
    - [x] (re-)installed with ip@4.3.0
- [x] Cheyenne (@climbfuji) - **not making any changes unless necessary, since machine will be decommissioned at the end of this year**
    - [x] external `bison` removed
    - ~~top-level directory structure~~
    - ~~3rdparty libraries and modules~~
    - [x] source cache location
    - [x] Installed in `/glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.5.0/envs/unified-env` (`gcc` only) and `/glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.5.0/envs/ufs-env` (`intel` only)
    - [x] basic tests performed for all supported compilers:
        - [x] jedi-bundle with gnu: all ctests pass
        - [x] ufs-weather-model with intel: I compiled as follows after loading `ufs-weather-model-env/1.0.0`, and that worked (no more testing done)
```
git clone -b feature/spack_stack_150 --recurse-submodules https://github.com/climbfuji/ufs-weather-model
mkdir build; cd build
cmake -DAPP=S2SWA -DCCPP_SUITES=FV3_GFS_v17_coupled_p8 ../ufs-weather-model 2>&1 | tee log.cmake
```

    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/787
    - [x] (re-)installed with ip@4.3.0
- [x] Derecho (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (jedi-bundle/intel)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/777
    - [x] (re-)installed with ip@4.3.0
```
The following tests FAILED:
        412 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        1403 - ufo_test_tier1_test_ufo_conventionalprofileprocessing_average_relativehumidity_obsfilter (Failed)
        1404 - ufo_test_tier1_test_ufo_conventionalprofileprocessing_average_relativehumidity_OPScomparison (Failed)
        1554 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
        1576 - fv3jedi_test_tier1_4denvar_seq (Failed)
        1587 - fv3jedi_test_tier1_diffstates_gfs (Failed)
        1650 - test_soca_sqrtvertloc (Failed)
```
[ctest.noget.timeout-0600.log](https://github.com/JCSDA/spack-stack/files/12729051/ctest.noget.timeout-0600.log)

- [x] Acorn (@AlexanderRichert-NOAA)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Install of unified-env for supported compilers (combined or separately) in `/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.5.0/envs/unified-env/`
    - [x] basic tests performed for all supported compilers (one or two UFS RTs)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/775
    - [x] (re-)installed with ip@4.3.0
- [x] Gaea C4 (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure **(aligned with C5 install!)**
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/lustre/f2/dev/wpo/role.epic/contrib/spack-stack/c4/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers: built jedi-bundle with Intel, ran ctests
    - [x] site/config and documentation updated: #777 
    - [x] (re-)installed with ip@4.3.0
```
Many fv3-jedi and soca ctests fail when run consecutively, but when run individually they pass. Very weird.
```

- [x Gaea C5 (@ulmononian, @climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/lustre/f2/dev/wpo/role.epic/contrib/spack-stack/c5/spack-stack-1.5.0/envs/unified-env`
    - [ ] ~~basic tests performed for all supported compilers~~
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/777
    - [x] (re-)installed with ip@4.3.0
- [x] Hera (@ulmononian)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Install of unified-env for supported compilers (combined or separately) in `/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (ufs-weather-model)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/779
    - [x] (re-)installed with ip@4.3.0
- [x] Jet (@ulmononian)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Install of unified-env for supported compilers (combined or separately) in `/mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (ufs-weather-model)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/779
    - [x] (re-)installed with ip@4.3.0
- [x] Narwhal (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in: `/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.5.0/envs/unified-env-intel-2021.4.0` and `/p/app/projects/NEPTUNE/spack-stack/spack-stack-1.5.0/envs/unified-env-gcc-10.3.0`
    - [x] basic tests performed for all supported compilers (jedi-bundle with Intel; basically all soca tests fail, the rest looks good)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/770
    - [x] (re-)installed with ip@4.3.0
```
         64 - test_l95_pseudomodel_state4d (Failed)
        403 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        1021 - ufo_test_tier1_instrument_sonde_gfs_HofX (Failed)
        1022 - ufo_test_tier1_instrument_sonde_gfs_qc (Failed)
        1508 - fv3jedi_test_tier1_linearmodel_htlm_physics (Failed)
        1577 - fv3jedi_test_tier1_eda_3dvar (Failed)
        1581 - test_soca_gridgen (Failed)
        1582 - test_soca_geometry (Failed)
        1583 - test_soca_geometry_iterator_2d (Failed)
        1584 - test_soca_geometry_iterator_3d (Failed)
        1585 - test_soca_state (Failed)
        1586 - test_soca_increment (Failed)
        1587 - test_soca_model (Failed)
        1588 - test_soca_modelaux (Failed)
        1589 - test_soca_getvalues (Failed)
        1590 - test_soca_errorcovariance (Failed)
        1591 - test_soca_linearmodel (Failed)
        1592 - test_soca_varchange_ana2model (Failed)
        1593 - test_soca_varchange_balance (Failed)
        1594 - test_soca_varchange_balance_TSSSH (Failed)
        1597 - test_soca_varchange_bkgerrsoca (Failed)
        1598 - test_soca_varchange_bkgerrsoca_stddev (Failed)
        1599 - test_soca_varchange_bkgerrgodas (Failed)
        1600 - test_soca_varchange_vertconv (Failed)
        1601 - test_soca_obslocalization (Failed)
        1605 - test_soca_forecast_mom6 (Failed)
        1607 - test_soca_forecast_mom6_ens1 (Failed)
        1608 - test_soca_forecast_mom6_ens2 (Failed)
        1610 - test_soca_static_socaerror_init (Failed)
        1612 - test_soca_setcorscales (Failed)
        1614 - test_soca_parameters_bump_cor_nicas_scales (Failed)
        1615 - test_soca_parameters_bump_loc (Failed)
        1617 - test_soca_enspert (Failed)
        1618 - test_soca_convertstate (Failed)
        1619 - test_soca_convertstate_changevar (Failed)
        1620 - test_soca_ensvariance (Failed)
        1621 - test_soca_ensmeanandvariance (Failed)
        1622 - test_soca_parametric_stddev (Failed)
        1623 - test_soca_ensrecenter (Failed)
        1624 - test_soca_hybridgain (Failed)
        1625 - test_soca_sqrtvertloc (Failed)
        1626 - test_soca_diffstates (Failed)
        1627 - test_soca_dirac_soca_cor_nicas_scales (Failed)
        1628 - test_soca_dirac_soca_cov (Failed)
        1629 - test_soca_dirac_socahyb_cov (Failed)
        1630 - test_soca_dirac_horizfilt (Failed)
        1631 - test_soca_dirac_soca_mask (Failed)
        1633 - test_soca_makeobs (Failed)
        1634 - test_soca_hofx_3d (Failed)
        1635 - test_soca_hofx_4d (Failed)
        1636 - test_soca_hofx_4d_pseudo (Failed)
        1637 - test_soca_enshofx (Failed)
        1638 - test_soca_3dvar_soca (Failed)
        1640 - test_soca_3dvar_godas (Failed)
        1641 - test_soca_3dvarlowres_soca (Failed)
        1642 - test_soca_3dvarfgat (Failed)
        1643 - test_soca_3dvarfgat_pseudo (Failed)
        1644 - test_soca_3dhyb (Failed)
        1645 - test_soca_3dhybfgat (Failed)
        1646 - test_soca_4denvar (Failed)
```

- [x] Nautilus (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in /p/app/projects/NEPTUNE/spack-stack/spack-stack-1.5.0/envs/unified-env
    - [x] basic tests performed for all supported compilers (jedi-bundle)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/775
    - [x] (re-)installed with ip@4.3.0
```
The following tests FAILED:
        407 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        1639 - test_soca_sqrtvertloc (Failed)
Errors while running CTest
```

- [x] S4 (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/data/prod/jedi/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed (jedi-bundle with intel)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/775
    - [x] (re-)installed with ip@4.3.0
```
The following tests FAILED:
        408 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        1462 - test_femps_csgrid (Failed)
        1541 - fv3jedi_test_tier1_dirac_geos_gsi_global (Failed)
        1542 - fv3jedi_test_tier1_3dvar_geos_sondes (Failed)
        1543 - fv3jedi_test_tier1_3dvar_geos_ozone (Failed)
        1544 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
        1545 - fv3jedi_test_tier1_3dvar_gfs_sondes (Failed)
        1627 - test_soca_setcorscales (Failed)
        1629 - test_soca_parameters_bump_cor_nicas_scales (Failed)
        1632 - test_soca_enspert (Failed)
        1640 - test_soca_sqrtvertloc (Failed)
        1642 - test_soca_dirac_soca_cor_nicas_scales (Failed)
        1644 - test_soca_dirac_socahyb_cov (Failed)
        1646 - test_soca_dirac_soca_mask (Failed)
        1647 - test_soca_dirac_soca_nomask (Failed)
        1655 - test_soca_3dvar_godas (Failed)
```

- [x] AWS JCSDA AMI Red Hat 8 (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] ~~3rdparty libraries and modules~~ (n/a)
    - [x] ~~source cache location~~ (n/a)
    - [x] Install of unified-env in `/home/ec2-user/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers: jedi bundle with gcc@11, ran several skylab experiments that fit on a single node, no test failures
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/787
    - [x] (re-)installed with ip@4.3.0
    - [x] AMI created in JCSDA USAF account (us-east-2), copied to us-east-1, made public
         - [x] us-east-1: [ami-059d445a90ad8b792](https://us-east-1.console.aws.amazon.com/ec2/home?region=us-east-1#ImageDetails:imageId=ami-059d445a90ad8b792)
         - [x] us-east-2: [ami-01fcf5d75ced5a046](https://us-east-2.console.aws.amazon.com/ec2/v2/home?region=us-east-2#ImageDetails:imageId=ami-01fcf5d75ced5a046)
- [x] AWS JCSDA AMI Ubuntu 20 (@climbfuji) - **support will be removed**
- [x] AWS JCSDA Parallel Cluster R&D Ubuntu 20, `hpc6a.48xlarge` instances (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/mnt/experiments-efs/skylab-v6/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (jedi-bundle with Intel and GNU)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/775
    - [x] (re-)installed with ip@4.3.0
```
The following tests FAILED with Intel (two weeks ago, 20230910):
	406 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
	1636 - test_soca_sqrtvertloc (Failed)
Errors while running **CTest

The following tests FAILED with GNU (today, 20230922):
	408 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
	1493 - fv3jedi_test_tier1_convertstate_gfs (Timeout)
	1494 - fv3jedi_test_tier1_convertstate_geos (Timeout)
	1562 - fv3jedi_test_tier1_hyb-fgat_fv3lm (Timeout)
	1641 - test_soca_sqrtvertloc (Failed)
```

- [x] AWS JCSDA Parallel Cluster ROMEX Ubuntu 20, `c6i.32xlarge` instances (@climbfuji)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/mnt/experiments-efs/ROMEX/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (jedi-bundle)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/775
    - [x] (re-)installed with ip@4.3.0
- [x] NOAA Parallel Works AWS (JCSDA - ** separately for EPIC for now? or sync?**)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/contrib/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (jedi-bundle intel)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/777
    - [x] (re-)installed with ip@4.3.0
    - [x] binary cache updated in JCSDA S3 bucket
```
The following tests FAILED:
        407 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        1140 - ufo_test_tier1_test_ufo_gmi_qc_filters (Failed)
        1141 - ufo_test_tier1_test_ufo_gmi_qc_filters_geos (Failed)
        1356 - ufo_test_tier1_test_ufo_opr_gmi_crtm (Failed)
        1541 - fv3jedi_test_tier1_dirac_geos_gsi_global (Failed)
        1542 - fv3jedi_test_tier1_3dvar_geos_sondes (Failed)
        1543 - fv3jedi_test_tier1_3dvar_geos_ozone (Failed)
        1544 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
        1545 - fv3jedi_test_tier1_3dvar_gfs_sondes (Failed)
        1551 - fv3jedi_test_tier1_hyb-3dvar (Failed)
        1576 - fv3jedi_test_tier1_diffstates_gfs (Failed)
        1579 - fv3jedi_test_tier1_addincrement_gfs (Failed)
        1626 - test_soca_setcorscales (Failed)
        1628 - test_soca_parameters_bump_cor_nicas_scales (Failed)
        1631 - test_soca_enspert (Failed)
        1639 - test_soca_sqrtvertloc (Failed)
        1641 - test_soca_dirac_soca_cor_nicas_scales (Failed)
        1643 - test_soca_dirac_socahyb_cov (Failed)
        1645 - test_soca_dirac_soca_mask (Failed)
        1646 - test_soca_dirac_soca_nomask (Failed)
        1654 - test_soca_3dvar_godas (Failed)
```

- [x] NOAA Parallel Works Azure (JCSDA - ** separately for EPIC for now? or sync?**)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/contrib/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (jedi-bundle intel)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/777
    - [x] (re-)installed with ip@4.3.0
    - [x] binary cache updated in JCSDA S3 bucket
```
The following tests FAILED:
        408 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        808 - test_ioda_obsspace (Failed)
        809 - test_ioda_obsspace_reader_pool (Failed)
        811 - test_ioda_obsspace_out_odc (Failed)
        812 - test_ioda_obsspace_out_odc_mpi_2 (Failed)
        813 - test_ioda_obsspace_out_odc_compare_aircraft (Failed)
        814 - test_ioda_obsspace_out_odc_compare_atms (Failed)
        815 - test_ioda_obsspace_out_odc_compare_gmi (Failed)
        816 - test_ioda_obsspace_out_odc_compare_amsua (Failed)
        817 - test_ioda_obsspace_out_odc_compare_sonde (Failed)
        818 - test_ioda_obsspace_out_odc_compare_aod (Failed)
        819 - test_ioda_obsspace_out_odc_compare_iasi (Failed)
        820 - test_ioda_obsspace_out_odc_compare_aod_testdatsets_0000 (Failed)
        821 - test_ioda_obsspace_out_odc_compare_aod_testdatsets_0001 (Failed)
        1019 - ufo_test_tier1_instrument_aircraft_gfs_HofX (Failed)
        1325 - ufo_test_tier1_test_ufo_opr_sfcpcorrected (Failed)
        1541 - fv3jedi_test_tier1_dirac_geos_gsi_global (Failed)
        1542 - fv3jedi_test_tier1_3dvar_geos_sondes (Failed)
        1543 - fv3jedi_test_tier1_3dvar_geos_ozone (Failed)
        1544 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
        1545 - fv3jedi_test_tier1_3dvar_gfs_sondes (Failed)
        1551 - fv3jedi_test_tier1_hyb-3dvar (Failed)
        1566 - fv3jedi_test_tier1_4denvar_seq (Failed)
        1577 - fv3jedi_test_tier1_diffstates_gfs (Failed)
        1580 - fv3jedi_test_tier1_addincrement_gfs (Failed)
        1627 - test_soca_setcorscales (Failed)
        1629 - test_soca_parameters_bump_cor_nicas_scales (Failed)
        1632 - test_soca_enspert (Failed)
        1640 - test_soca_sqrtvertloc (Failed)
        1642 - test_soca_dirac_soca_cor_nicas_scales (Failed)
        1644 - test_soca_dirac_socahyb_cov (Failed)
        1646 - test_soca_dirac_soca_mask (Failed)
        1647 - test_soca_dirac_soca_nomask (Failed)
        1655 - test_soca_3dvar_godas (Failed)
```

- [x] NOAA Parallel Works Gcloud (JCSDA - ** separately for EPIC for now? or sync?**)
    - [x] external `bison` removed
    - [x] top-level directory structure
    - [x] 3rdparty libraries and modules
    - [x] source cache location
    - [x] Installed in `/contrib/spack-stack/spack-stack-1.5.0/envs/unified-env`
    - [x] basic tests performed for all supported compilers (jedi-bundle intel)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/777
    - [x] (re-)installed with ip@4.3.0
    - [x] binary cache updated in JCSDA S3 bucket
```
The following tests FAILED:
        407 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        1541 - fv3jedi_test_tier1_dirac_geos_gsi_global (Failed)
        1542 - fv3jedi_test_tier1_3dvar_geos_sondes (Failed)
        1543 - fv3jedi_test_tier1_3dvar_geos_ozone (Failed)
        1544 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
        1545 - fv3jedi_test_tier1_3dvar_gfs_sondes (Failed)
        1551 - fv3jedi_test_tier1_hyb-3dvar (Failed)
        1576 - fv3jedi_test_tier1_diffstates_gfs (Failed)
        1579 - fv3jedi_test_tier1_addincrement_gfs (Failed)
        1626 - test_soca_setcorscales (Failed)
        1628 - test_soca_parameters_bump_cor_nicas_scales (Failed)
        1631 - test_soca_enspert (Failed)
        1639 - test_soca_sqrtvertloc (Failed)
        1641 - test_soca_dirac_soca_cor_nicas_scales (Failed)
        1643 - test_soca_dirac_socahyb_cov (Failed)
        1645 - test_soca_dirac_soca_mask (Failed)
        1646 - test_soca_dirac_soca_nomask (Failed)
        1654 - test_soca_3dvar_godas (Failed)
```

- [x] JCSDA CI container `gnu/openmpi`, `clang/mpich`, `intel/impi`
    - [x] external `bison` removed
    - [x] top-level directory structure (n/a)
    - [x] 3rdparty libraries and modules (n/a)
    - [x] source cache location (n/a)
    - [x] (re-)built with ip@4.3.0
    - [x] basic tests performed (JCSDA CI)
    - [x] Uploaded to JCSDA NOAA AWS ECR as "latest"
    - [x] Uploaded to dockerhub as "latest" (not Intel due to licensing restrictions)
    - [x] Converted to Singularity and signed
    - [x] Singularity containers uploaded to JCSDA S3 and to sylabs.io (the latter not for Intel due to licensing restrictions)
    - [x] site/config and documentation updated: https://github.com/JCSDA/spack-stack/pull/775
```
The following tests FAILED:
        407 - saber_test_dirac_gsi_gfs_global_1-1 (Failed)
        520 - get_crtm_coeffs (Failed)
        1541 - fv3jedi_test_tier1_dirac_geos_gsi_global (Failed)
        1542 - fv3jedi_test_tier1_3dvar_geos_sondes (Failed)
        1543 - fv3jedi_test_tier1_3dvar_geos_ozone (Failed)
        1544 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
        1545 - fv3jedi_test_tier1_3dvar_gfs_sondes (Failed)
        1626 - test_soca_setcorscales (Failed)
        1628 - test_soca_parameters_bump_cor_nicas_scales (Failed)
        1631 - test_soca_enspert (Failed)
        1639 - test_soca_sqrtvertloc (Failed)
        1641 - test_soca_dirac_soca_cor_nicas_scales (Failed)
        1643 - test_soca_dirac_socahyb_cov (Failed)
        1645 - test_soca_dirac_soca_mask (Failed)
        1646 - test_soca_dirac_soca_nomask (Failed)
        1654 - test_soca_3dvar_godas (Failed)
```

on hera using spack-stack/1.5.0, the cmake step for the s2swa config of the ufs-wm fails:

```
-- Configuring done
CMake Error at /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.0/envs/unified-env/install/intel/2021.5.0/mapl-2.35.2-rwaq27r/lib64/cmake/MAPL/MAPL-targets.cmake:158 (set_target_properties):
  The link interface of target "MAPL.gridcomps" contains:

    FARGPARSE::fargparse

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.0/envs/unified-env/install/intel/2021.5.0/mapl-2.35.2-rwaq27r/lib64/cmake/MAPL/mapl-config.cmake:74 (include)
  GOCART/CMakeLists.txt:64 (find_package)


CMake Error at /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.0/envs/unified-env/install/intel/2021.5.0/mapl-2.35.2-rwaq27r/lib64/cmake/MAPL/MAPL-targets.cmake:166 (set_target_properties):
  The link interface of target "MAPL.cap" contains:

    FARGPARSE::fargparse

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.0/envs/unified-env/install/intel/2021.5.0/mapl-2.35.2-rwaq27r/lib64/cmake/MAPL/mapl-config.cmake:74 (include)
  GOCART/CMakeLists.txt:64 (find_package)


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/configs/sites/casper/compilers.yaml
+++ b/configs/sites/casper/compilers.yaml
@@ -1,23 +1,45 @@
 compilers::
+  #- compiler:
+  #    spec: intel@19.1.1.217
+  #    paths:
+  #      cc: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/icc
+  #      cxx: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/icpc
+  #      f77: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/ifort
+  #      fc: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/ifort
+  #    flags: {}
+  #    operating_system: centos7
+  #    target: x86_64
+  #    modules:
+  #    - intel/19.1.1
+  #    environment:
+  #      prepend_path:
+  #        PATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/bin'
+  #        CPATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/include'
+  #        LD_LIBRARY_PATH: '/glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/lib/intel64_lin:/glade/u/apps/opt/intel/2020u1/compilers_and_libraries_2020.1.217/linux/mpi/intel64/libfabric/lib:/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/lib64'
+  #      set:
+  #        INTEL_LICENSE_FILE: '28518@128.117.177.41'
+  #        LM_LICENSE_FILE: '28518@128.117.177.41'
+  #        I_MPI_ROOT: '/glade/u/apps/opt/intel/2020u1/impi/2019.7.217/intel64'
+  #    extra_rpaths: []
   - compiler:
-      spec: intel@19.1.1.217
+      spec: intel@2022.0.2
       paths:
-        cc: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/icc
-        cxx: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/icpc
-        f77: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/ifort
-        fc: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/ifort
+        cc: /glade/u/apps/opt/intel/2022.1/compiler/2022.0.2/linux/bin/intel64/icc
+        cxx: /glade/u/apps/opt/intel/2022.1/compiler/2022.0.2/linux/bin/intel64/icpc
+        f77: /glade/u/apps/opt/intel/2022.1/compiler/2022.0.2/linux/bin/intel64/ifort
+        fc: /glade/u/apps/opt/intel/2022.1/compiler/2022.0.2/linux/bin/intel64/ifort
       flags: {}
       operating_system: centos7
       target: x86_64
       modules:
-      - intel/19.1.1
+      - intel/2022.1
       environment:
         prepend_path:
           PATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/bin'
           CPATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/include'
-          LD_LIBRARY_PATH: '/glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/lib/intel64_lin:/glade/u/apps/opt/intel/2020u1/compilers_and_libraries_2020.1.217/linux/mpi/intel64/libfabric/lib:/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/lib64'
+          LD_LIBRARY_PATH: '/glade/u/apps/opt/intel/2022.1/compiler/latest/linux/compiler/lib/intel64_lin:/glade/u/apps/opt/intel/2022.1/mpi/2021.5.1/libfabric/lib:/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/lib64'
         set:
           INTEL_LICENSE_FILE: '28518@128.117.177.41'
           LM_LICENSE_FILE: '28518@128.117.177.41'
-          I_MPI_ROOT: '/glade/u/apps/opt/intel/2020u1/impi/2019.7.217/intel64'
+          #I_MPI_ROOT: '/glade/u/apps/opt/intel/2022.1/mpi/2021.5.1'
       extra_rpaths: []

--- a/configs/sites/casper/compilers.yaml
+++ b/configs/sites/casper/compilers.yaml
@@ -13,9 +13,9 @@ compilers::
       - intel/19.1.1
       environment:
         prepend_path:
-          PATH: '/glade/u/apps/dav/opt/gnu/9.1.0/bin'
-          CPATH: '/glade/u/apps/dav/opt/gnu/9.1.0/include'
-          LD_LIBRARY_PATH: '/glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/lib/intel64_lin:/glade/u/apps/opt/intel/2020u1/compilers_and_libraries_2020.1.217/linux/mpi/intel64/libfabric/lib:/glade/u/apps/dav/opt/gnu/9.1.0/lib64'
+          PATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/bin'
+          CPATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/include'
+          LD_LIBRARY_PATH: '/glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/lib/intel64_lin:/glade/u/apps/opt/intel/2020u1/compilers_and_libraries_2020.1.217/linux/mpi/intel64/libfabric/lib:/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/lib64'
         set:
           INTEL_LICENSE_FILE: '28518@128.117.177.41'
           LM_LICENSE_FILE: '28518@128.117.177.41'

--- a/configs/sites/casper/compilers.yaml
+++ b/configs/sites/casper/compilers.yaml
@@ -1,26 +1,4 @@
 compilers::
-  #- compiler:
-  #    spec: intel@19.1.1.217
-  #    paths:
-  #      cc: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/icc
-  #      cxx: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/icpc
-  #      f77: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/ifort
-  #      fc: /glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/bin/intel64/ifort
-  #    flags: {}
-  #    operating_system: centos7
-  #    target: x86_64
-  #    modules:
-  #    - intel/19.1.1
-  #    environment:
-  #      prepend_path:
-  #        PATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/bin'
-  #        CPATH: '/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/include'
-  #        LD_LIBRARY_PATH: '/glade/u/apps/opt/intel/2020u1/compilers_and_libraries/linux/lib/intel64_lin:/glade/u/apps/opt/intel/2020u1/compilers_and_libraries_2020.1.217/linux/mpi/intel64/libfabric/lib:/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/lib64'
-  #      set:
-  #        INTEL_LICENSE_FILE: '28518@128.117.177.41'
-  #        LM_LICENSE_FILE: '28518@128.117.177.41'
-  #        I_MPI_ROOT: '/glade/u/apps/opt/intel/2020u1/impi/2019.7.217/intel64'
-  #    extra_rpaths: []
   - compiler:
       spec: intel@2022.0.2
       paths:
@@ -41,5 +19,6 @@ compilers::
         set:
           INTEL_LICENSE_FILE: '28518@128.117.177.41'
           LM_LICENSE_FILE: '28518@128.117.177.41'
-          #I_MPI_ROOT: '/glade/u/apps/opt/intel/2022.1/mpi/2021.5.1'
+          I_MPI_ROOT: '/glade/u/apps/opt/intel/2022.1/mpi/2021.5.1'
+
       extra_rpaths: []

--- a/configs/sites/casper/modules.yaml
+++ b/configs/sites/casper/modules.yaml
@@ -5,3 +5,5 @@ modules:
     lmod:
       exclude:
       - ecflow
+      include:
+      - python

--- a/configs/sites/casper/packages.yaml
+++ b/configs/sites/casper/packages.yaml
@@ -1,18 +1,25 @@
 packages:
   all:
-    compiler:: [intel@19.1.1.217]
+    #compiler:: [intel@19.1.1.217]
+    compiler:: [intel@2022.0.2]
     providers:
-      mpi:: [intel-mpi@2019.7.217]
+      #mpi:: [intel-mpi@2019.7.217]
+      mpi:: [intel-oneapi-mpi@2021.5.1]
 
 ### MPI, Python, MKL
   mpi:
     buildable: False
-  intel-mpi:
+  #intel-mpi:
+  #  externals:
+  #  - spec: intel-mpi@2019.7.217%intel@19.1.1.217
+  #    prefix: /glade/u/apps/opt/intel/2020u1
+  #    modules:
+  #    - impi/2019.7.217
+  intel-oneapi-mpi:
     externals:
-    - spec: intel-mpi@2019.7.217%intel@19.1.1.217
-      prefix: /glade/u/apps/opt/intel/2020u1
+    - spec: intel-oneapi-mpi@2021.5.1%intel@2022.0.2
       modules:
-      - impi/2019.7.217
+      - impi/2022.1
 
 ### All other external packages listed alphabetically
   autoconf:
@@ -31,10 +38,6 @@ packages:
     externals:
     - spec: binutils@2.27.43
       prefix: /usr
-  #bison:
-  #  externals:
-  #  - spec: bison@3.0.4
-  #    prefix: /usr
   bzip2:
     externals:
     - spec: bzip2@1.0.6
@@ -73,11 +76,6 @@ packages:
       prefix: /glade/work/epicufsrt/contrib/spack-stack/casper/ecflow-5.8.4
       modules:
       - ecflow/5.8.4
-  # Don't use, leads to duplicate packages being built
-  #expat:
-  #  externals:
-  #  - spec: expat@1.6.0
-  #    prefix: /usr
   file:
     externals:
     - spec: file@5.11
@@ -168,8 +166,8 @@ packages:
     - spec: openssl@1.0.2k-fips
       prefix: /usr
   # Pin patchelf to 0.15.0 for Intel compiler (no C++-17 features)
-  patchelf:
-    version:: ['0.15.0']
+  #patchelf:
+  #  version:: ['0.15.0']
   perl:
     externals:
     - spec: perl@5.16.3~cpanm+shared+threads

--- a/configs/sites/casper/packages.yaml
+++ b/configs/sites/casper/packages.yaml
@@ -13,20 +13,6 @@ packages:
       prefix: /glade/u/apps/opt/intel/2020u1
       modules:
       - impi/2019.7.217
-  python:
-    buildable: False
-    externals:
-    - spec: python@3.9.12+bz2+ctypes+dbm+lzma+nis+pyexpat+pythoncmd+readline+sqlite3+ssl+tix+tkinter+uuid+zlib
-      prefix: /glade/work/jedipara/cheyenne/spack-stack/miniconda-3.9.12
-      modules:
-      - miniconda/3.9.12
-  py-pip:
-    buildable: False
-    externals:
-    - spec: py-pip@21.2.4
-      prefix: /glade/work/jedipara/cheyenne/spack-stack/miniconda-3.9.12
-      modules:
-      - miniconda/3.9.12
 
 ### All other external packages listed alphabetically
   autoconf:
@@ -45,10 +31,10 @@ packages:
     externals:
     - spec: binutils@2.27.43
       prefix: /usr
-  bison:
-    externals:
-    - spec: bison@3.0.4
-      prefix: /usr
+  #bison:
+  #  externals:
+  #  - spec: bison@3.0.4
+  #    prefix: /usr
   bzip2:
     externals:
     - spec: bzip2@1.0.6
@@ -84,7 +70,7 @@ packages:
     buildable: False
     externals:
     - spec: ecflow@5.8.4+ui+static_boost
-      prefix: /glade/work/jedipara/cheyenne/spack-stack/ecflow-5.8.4
+      prefix: /glade/work/epicufsrt/contrib/spack-stack/casper/ecflow-5.8.4
       modules:
       - ecflow/5.8.4
   # Don't use, leads to duplicate packages being built
@@ -163,7 +149,7 @@ packages:
     buildable: False
     externals:
     - spec: mysql@8.0.31
-      prefix: /glade/work/jedipara/cheyenne/spack-stack/mysql-8.0.31
+      prefix: /glade/work/epicufsrt/contrib/spack-stack/casper/mysql-8.0.31
       modules:
       - mysql/8.0.31
   ncurses:
@@ -195,7 +181,7 @@ packages:
   qt:
     externals:
     - spec: qt@5.15.2
-      prefix: /glade/work/jedipara/cheyenne/spack-stack/qt-5.15.2/5.15.2/gcc_64
+      prefix: /glade/work/epicufsrt/contrib/spack-stack/casper/qt-5.15.2
   rsync:
     externals:
     - spec: rsync@3.1.2
@@ -207,10 +193,6 @@ packages:
   sed:
     externals:
     - spec: sed@4.2.2
-      prefix: /usr
-  sqlite:
-    externals:
-    - spec: sqlite@3.8.10.2~fts~functions+rtree
       prefix: /usr
   subversion:
     externals:

--- a/configs/sites/casper/packages.yaml
+++ b/configs/sites/casper/packages.yaml
@@ -1,25 +1,23 @@
 packages:
   all:
-    #compiler:: [intel@19.1.1.217]
     compiler:: [intel@2022.0.2]
     providers:
-      #mpi:: [intel-mpi@2019.7.217]
       mpi:: [intel-oneapi-mpi@2021.5.1]
 
 ### MPI, Python, MKL
   mpi:
     buildable: False
-  #intel-mpi:
-  #  externals:
-  #  - spec: intel-mpi@2019.7.217%intel@19.1.1.217
-  #    prefix: /glade/u/apps/opt/intel/2020u1
-  #    modules:
-  #    - impi/2019.7.217
   intel-oneapi-mpi:
     externals:
     - spec: intel-oneapi-mpi@2021.5.1%intel@2022.0.2
+      prefix: /glade/u/apps/opt/intel/2022.1
       modules:
-      - impi/2022.1
+      - intel-oneapi-mpi/2021.5.1.lua
+
+### Modification of common packages
+  # Default gdal (3.7.0) doesn't build on Casper with intel@2022.0.2
+  gdal:
+    version:: ['3.6.3']
 
 ### All other external packages listed alphabetically
   autoconf:

--- a/configs/sites/cheyenne/compilers.yaml
+++ b/configs/sites/cheyenne/compilers.yaml
@@ -54,6 +54,8 @@ compilers::
       target: x86_64
       modules:
       - gnu/10.1.0
-      environment: {}
+      environment:
+        set:
+          OMPI_MCA_rmaps_base_oversubscribe: '1'
       extra_rpaths: []
 

--- a/configs/sites/cheyenne/packages.yaml
+++ b/configs/sites/cheyenne/packages.yaml
@@ -40,6 +40,20 @@ packages:
       prefix: /glade/u/apps/ch/opt/openmpi/4.1.1/gnu/10.1.0
       modules:
       - openmpi/4.1.1
+  python:
+    buildable: False
+    externals:
+    - spec: python@3.9.12+bz2+ctypes+dbm+lzma+nis+pyexpat+pythoncmd+readline+sqlite3+ssl+tix+tkinter+uuid+zlib
+      prefix: /glade/work/jedipara/cheyenne/spack-stack/miniconda-3.9.12
+      modules:
+      - miniconda/3.9.12
+  py-pip:
+    buildable: False
+    externals:
+    - spec: py-pip@21.2.4
+      prefix: /glade/work/jedipara/cheyenne/spack-stack/miniconda-3.9.12
+      modules:
+      - miniconda/3.9.12
 
 ### All other external packages listed alphabetically
   autoconf:
@@ -177,6 +191,10 @@ packages:
   openssh:
     externals:
     - spec: openssh@7.2p2
+      prefix: /usr
+  openssl:
+    externals:
+    - spec: openssl@1.0.2p-fips
       prefix: /usr
   # Pin patchelf to 0.15.0 for Intel compiler (no C++-17 features)
   patchelf:

--- a/doc/modulefile_templates/gcc
+++ b/doc/modulefile_templates/gcc
@@ -1,0 +1,23 @@
+#%Module1.0
+
+module-whatis "Provides a gcc-10.1.0 installation for use with spack."
+
+conflict gnu
+conflict gcc
+
+proc ModulesHelp { } {
+puts stderr "Provides a gcc-10.1.0 installation for use with spack."
+}
+
+
+# Set this value
+set GCC_PATH "/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0"
+
+prepend-path PATH "${GCC_PATH}/bin"
+prepend-path LD_LIBRARY_PATH "${GCC_PATH}/lib"
+prepend-path LD_LIBRARY_PATH "${GCC_PATH}/lib64"
+prepend-path LIBRARY_PATH "${GCC_PATH}/lib"
+prepend-path LIBRARY_PATH "${GCC_PATH}/lib64"
+prepend-path CPATH "${GCC_PATH}/include"
+prepend-path CMAKE_PREFIX_PATH "${GCC_PATH}"
+prepend-path MANPATH "${GCC_PATH}/share/man"

--- a/doc/source/MaintainersSection.rst
+++ b/doc/source/MaintainersSection.rst
@@ -429,6 +429,9 @@ gcc
    make install 2>&1 | tee log.install
    # create modulefile
 
+intel-oneapi-mpi (module only)
+   The ``impi/2022.1`` module provided by CISL has the wrong ``I_MPI_ROOT`` and ``MPI_ROOT`` values. Copy the file to ``/glade/work/epicufsrt/contrib/spack-stack/casper/modulefiles/intel-oneapi-mpi/2021.5.1.lua`` and remove the offending entries, as well as the unnecessary ``ncarcompilers`` logic at the end.
+
 qt (qt@5)
    The default ``qt@5`` in ``/usr`` is incomplete and thus insufficient for building ``ecflow``. Follow these instructions to build ``qt@5.15.2`` using ``gcc@10.10.0``. See also https://wiki.qt.io/Building_Qt_5_from_Git#Getting_the_source_code for building qt from source.
 

--- a/doc/source/MaintainersSection.rst
+++ b/doc/source/MaintainersSection.rst
@@ -408,7 +408,67 @@ mysql
 NCAR-Wyoming Casper
 ------------------------------
 
-Casper is co-located with Cheyenne and shares the parallel filesystem ``/glade`` and more with it. It is, however, a different operating system with a somewhat different software stack. spack-stack was installed on Casper after it was installed on Cheyenne, and prerequisites from Cheyenne were reused where possible (``miniconda``, ``qt``, ``ecflow``, ``mysql``). See below for information on how to install these packages.
+On Casper, there are problems with newer versions of the Intel compiler/MPI library when trying to run MPI jobs with just one task (``mpiexec -np 1``) - for JEDI, job hangs forever in a particular MPI communication call in oops. This is why an older version Intel 19 is used here.
+
+gcc
+   CISL does not provide a newer GNU compiler for Casper, and the Cheyenne compiler has problems with missing symbols. Build ``gcc`` etc. as follows and create a module file from template ``gcc`` in ``/glade/work/epicufsrt/contrib/spack-stack/casper/modulefiles``.
+
+.. code-block:: console
+
+   module purge
+   mkdir -p /glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/src
+   cd /glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0/src
+   wget http://ftp.mirrorservice.org/sites/sourceware.org/pub/gcc/releases/gcc-10.1.0/gcc-10.1.0.tar.gz
+   tar -xvzf gcc-10.1.0.tar.gz
+   cd gcc-10.1.0/
+   ./contrib/download_prerequisites
+   ./configure \
+       --prefix=/glade/work/epicufsrt/contrib/spack-stack/casper/gcc-10.1.0 \
+       --disable-multilib --enable-languages=c,c++ 2>&1 | tee log.config
+   make -j8 2>&1 | tee log.make
+   make install 2>&1 | tee log.install
+   # create modulefile
+
+qt (qt@5)
+   The default ``qt@5`` in ``/usr`` is incomplete and thus insufficient for building ``ecflow``. Follow these instructions to build ``qt@5.15.2`` using ``gcc@10.10.0``. See also https://wiki.qt.io/Building_Qt_5_from_Git#Getting_the_source_code for building qt from source.
+
+.. code-block:: console
+
+   module purge
+   export LMOD_TMOD_FIND_FIRST=yes
+   module use /glade/work/epicufsrt/contrib/spack-stack/casper/modulefiles
+   module load gcc/10.1.0
+   mkdir -p /glade/work/epicufsrt/contrib/spack-stack/casper/qt-5.15.2/src
+   cd /glade/work/epicufsrt/contrib/spack-stack/casper/qt-5.15.2/src
+   git clone https://code.qt.io/qt/qt5.git
+   cd qt5/
+   git checkout 5.12
+   perl init-repository 2>&1 | tee log.init-repository
+   cd ..
+   mkdir qt5-build
+   cd qt5-build
+   ../qt5/configure -opensource -nomake examples -nomake tests \
+       -prefix "/glade/work/epicufsrt/contrib/spack-stack/casper/qt-5.15.2" 2>&1 | tee log.config
+   make -j4 2>&1 | tee log.make
+   make install 2>&1 | tee log.install
+   # If errors occur during the installation of qtlocation, ignore. This is one of the last steps
+   # and not needed for ecflow (consider not building this module in the first place ... todo).
+
+ecflow
+  ``ecFlow`` must be built manually using the GNU compilers and linked against a static ``boost`` library. After installing `qt5`, and loading the following modules, follow the instructions in :numref:`Section %s <MaintainersSection_ecFlow>`.
+
+.. code-block:: console
+
+   module purge
+   export LMOD_TMOD_FIND_FIRST=yes
+   module use /glade/work/epicufsrt/contrib/spack-stack/casper/modulefiles
+   module load gnu/10.1.0
+   module load python/3.7.9
+   module load qt/5.15.2
+   module load cmake/3.18.2
+
+mysql
+  ``mysql`` must be installed separately from ``spack`` using a binary tarball provided by the MySQL community. Follow the instructions in :numref:`Section %s <MaintainersSection_MySQL>` to install ``mysql`` in ``/glade/work/epicufsrt/contrib/spack-stack/casper/mysql-8.0.31``.
 
 .. _MaintainersSection_Cheyenne:
 

--- a/doc/source/NewSiteConfigs.rst
+++ b/doc/source/NewSiteConfigs.rst
@@ -453,7 +453,10 @@ It is recommended to increase the stacksize limit by using ``ulimit -S -s unlimi
 
 .. code-block:: console
 
-   spack external find --scope system --exclude bison --exclude cmake --exclude curl
+   spack external find --scope system \
+       --exclude bison --exclude cmake \
+       --exclude curl --exclude openssl \
+       --exclude openssh
    spack external find --scope system perl
    spack external find --scope system wget
    spack external find --scope system mysql
@@ -512,9 +515,6 @@ It is recommended to increase the stacksize limit by using ``ulimit -S -s unlimi
    - compilers: ['%gcc']
 
 11. Edit site config files and common config files, for example to remove duplicate versions of external packages that are unwanted, add specs in ``envs/unified-env.mylinux/spack.yaml``, etc.
-
-.. warning::
-   Remove any external ``cmake`` package version 3.20 or earlier from ``envs/unified-env.mylinux/site/packages.yaml`` (or use add ``--exclude cmake`` to the ``spack external find`` command). Further, on Red Hat/CentOS, remove any external ``curl`` that might have been found.
 
 .. code-block:: console
 

--- a/doc/source/NewSiteConfigs.rst
+++ b/doc/source/NewSiteConfigs.rst
@@ -453,13 +453,11 @@ It is recommended to increase the stacksize limit by using ``ulimit -S -s unlimi
 
 .. code-block:: console
 
-   spack external find --scope system # use '--exclude' for troublesome packages like bison@:3.3, openssl@1.1.1, and cmake 3.20 and earlier
+   spack external find --scope system --exclude bison --exclude cmake --exclude curl
    spack external find --scope system perl
    spack external find --scope system wget
    spack external find --scope system mysql
    spack external find --scope system texlive
-   # On Ubuntu (but not on Red Hat):
-   spack external find --scope system curl
 
 5. Find compilers, add to site config's ``compilers.yaml``
 

--- a/doc/source/PreConfiguredSites.rst
+++ b/doc/source/PreConfiguredSites.rst
@@ -24,7 +24,7 @@ Ready-to-use spack-stack 1.5.0 installations are available on the following, ful
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
 | NASA                | Discover                         | GCC, Intel      | ``/gpfsm/dswdev/jcsda/spack-stack/spack-stack-1.5.0/envs/unified-env``                                  | Dom Heinzeller / ???          |
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
-|                     | Casper                           | Intel           | ``/glade/work/epicufsrt/contrib/spack-stack/casper/spack-stack-1.4.1/envs/unified-env``                 | Dom Heinzeller / ???          |
+|                     | Casper                           | Intel           | ``/glade/work/epicufsrt/contrib/spack-stack/casper/spack-stack-1.5.0/envs/unified-env``                 | Dom Heinzeller / ???          |
 |                     +----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
 | NCAR-Wyoming        | Cheyenne                         | GCC, Intel      | ``/glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.5.0/envs/{unified-env,ufs-env}``     | Cam Book / Dom Heinzeller     |
 |                     +----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
@@ -315,14 +315,14 @@ The following is required for building new spack environments and for using spac
    module load ecflow/5.8.4
    module load mysql/8.0.31
 
-For ``spack-stack-1.4.1`` with Intel, load the following modules after loading miniconda and ecflow.
+For ``spack-stack-1.5.0`` with Intel, load the following modules after loading miniconda and ecflow.
 
 .. code-block:: console
 
-   module use /glade/work/epicufsrt/contrib/spack-stack/casper/spack-stack-1.4.1/envs/unified-env/install/modulefiles/Core
-   module load stack-intel/19.1.1.217
-   module load stack-intel-mpi/2019.7.217
-   module load stack-python/3.9.12
+   module use /glade/work/epicufsrt/contrib/spack-stack/casper/spack-stack-1.5.0/envs/unified-env/install/modulefiles/Core
+   module load stack-intel/2022.0.2
+   module load stack-intel-oneapi-mpi/2021.5.1
+   module load stack-python/3.10.8
    module available
 
 .. _Preconfigured_Sites_Cheyenne:
@@ -400,7 +400,7 @@ For ``spack-stack-1.5.0`` with Intel, load the following modules after loading e
 NOAA Acorn (WCOSS2 test system)
 -------------------------------
 
-For spack-stack-1.4.1, the meta modules are in ``/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.4.1/envs/unified-env/install/modulefiles/Core``.
+For spack-stack-1.5.0, the meta modules are in ``/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.5.0/envs/unified-env/install/modulefiles/Core``.
 
 On WCOSS2 OpenSUSE sets ``CONFIG_SITE`` which causes libraries to be installed in ``lib64``, breaking the ``lib`` assumption made by some packages. Therefore, ``CONFIG_SITE`` should be set to empty in ``compilers.yaml``. Also, don't use ``module purge`` on Acorn!
 
@@ -508,10 +508,6 @@ For ``spack-stack-1.5.0`` with Intel, load the following modules after loading m
    module load stack-cray-mpich/8.1.25
    module load stack-python/3.10.8
    module -t available
-
-.. note::
-
-   The recent update to ``spack-stack-dev-20230717`` was required on Gaea C5 due to a bug in the Intel compilers used in ``spack-stack-1.4.1``.
 
 .. note::
    On Gaea C5, running ``module available`` without the option ``-t`` leads to an error: ``/usr/bin/lua5.3: /opt/cray/pe/lmod/lmod/libexec/Spider.lua:568: stack overflow``
@@ -688,7 +684,7 @@ For ``spack-stack-1.5.0`` with Intel on the JCSDA ROMEX cluster (``c6i.32xlarge`
 Amazon Web Services Red Hat 8
 -----------------------------
 
-Use a c6i.4xlarge instance or larger if running out of memory with AMI "skylab-6.0.0-redhat8" (ami-MISSING in region us-east-1, ami-01fcf5d75ced5a046 in region us-east-2).
+Use a c6i.4xlarge instance or larger if running out of memory with AMI "skylab-6.0.0-redhat8" (ami-059d445a90ad8b792 in region us-east-1, ami-01fcf5d75ced5a046 in region us-east-2).
 
 For ``spack-stack-1.5.0``, run:
 
@@ -715,7 +711,7 @@ TACC Frontera
 ------------------------------
 
 .. note::
-   ``spack-stack-1.4.0`` is currently not supported on this platform and may be added in the near future.
+   ``spack-stack-1.5.0`` is currently not supported on this platform.
 
 The following is required for building new spack environments and for using spack to build and run software.
 

--- a/doc/source/PreConfiguredSites.rst
+++ b/doc/source/PreConfiguredSites.rst
@@ -26,7 +26,7 @@ Ready-to-use spack-stack 1.5.0 installations are available on the following, ful
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
 |                     | Casper                           | Intel           | ``/glade/work/epicufsrt/contrib/spack-stack/casper/spack-stack-1.4.1/envs/unified-env``                 | Dom Heinzeller / ???          |
 |                     +----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
-| NCAR-Wyoming        | Cheyenne                         | GCC, Intel      | ``/glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.4.1/envs/unified-env``               | Cam Book / Dom Heinzeller     |
+| NCAR-Wyoming        | Cheyenne                         | GCC, Intel      | ``/glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.5.0/envs/{unified-env,ufs-env}``     | Cam Book / Dom Heinzeller     |
 |                     +----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
 |                     | Derecho                          | Intel           | ``/glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.0/envs/unified-env``                | Mark Potts / Dom Heinzeller   |
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
@@ -54,11 +54,11 @@ Ready-to-use spack-stack 1.5.0 installations are available on the following, ful
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
 | **Cloud platforms**                                                                                                                                                                                                |
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
-|                     | AMI Red Hat 8                    | GCC             | ``/home/ec2-user/spack-stack/spack-stack-1.4.0/envs/unified-env``                                       | Dom Heinzeller / ???          |
+|                     | AMI Red Hat 8                    | GCC             | ``/home/ec2-user/spack-stack/spack-stack-1.5.0/envs/unified-env``                                       | Dom Heinzeller / ???          |
 +                     +----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
-| Amazon Web Services | Parallelcluster JCSDA R&D        |                 | ``/mnt/experiments-efs/skylab-v6/spack-stack-1.5.0/envs/unified-env``                                   | Dom Heinzeller / ???          |
+| Amazon Web Services | Parallelcluster JCSDA R&D        | Intel           | ``/mnt/experiments-efs/skylab-v6/spack-stack-1.5.0/envs/unified-env``                                   | Dom Heinzeller / ???          |
 +                     +----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
-|                     | Parallelcluster JCSDA ROMEX      |                 | ``/mnt/experiments-efs/ROMEX/spack-stack-1.5.0/envs/unified-env``                                       | Dom Heinzeller / ???          |
+|                     | Parallelcluster JCSDA ROMEX      | Intel           | ``/mnt/experiments-efs/ROMEX/spack-stack-1.5.0/envs/unified-env``                                       | Dom Heinzeller / ???          |
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
 | NOAA (RDHPCS)       | RDHPCS Cloud (Parallel Works)^** | Intel           | ``/contrib/spack-stack/spack-stack-1.5.0/envs/unified-env``                                             | Mark Potts / Cam Book / Dom H |
 +---------------------+----------------------------------+-----------------+---------------------------------------------------------------------------------------------------------+-------------------------------+
@@ -310,7 +310,7 @@ The following is required for building new spack environments and for using spac
 
    module purge
    export LMOD_TMOD_FIND_FIRST=yes
-   module use /glade/work/jedipara/cheyenne/spack-stack/modulefiles/misc
+   module use /glade/work/epicufsrt/contrib/spack-stack/casper/modulefiles
    module load python/3.7.9
    module load ecflow/5.8.4
    module load mysql/8.0.31
@@ -342,21 +342,21 @@ The following is required for building new spack environments and for using spac
    module load ecflow/5.8.4
    module load mysql/8.0.31
 
-For ``spack-stack-1.4.1`` with Intel, load the following modules after loading miniconda, ecflow and mysql.
+For ``spack-stack-1.5.0`` with Intel, please note that there is no support for the full unified environment (``unified-env``) due to the lack of support of the C++-17 standard in Intel 19. An environment for the Unified Forecast System is available by loading the following modules after loading miniconda, ecflow and mysql.
 
 .. code-block:: console
 
-   module use /glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.4.1/envs/unified-env
+   module use /glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.5.0/envs/ufs-env/install/modulefiles/Core
    module load stack-intel/19.1.1.217
    module load stack-intel-mpi/2019.7.217
    module load stack-python/3.9.12
    module available
 
-For ``spack-stack-1.4.1`` with GNU, load the following modules after loading miniconda, ecflow and mysql.
+For ``spack-stack-1.5.0`` with GNU, load the following modules after loading miniconda, ecflow and mysql. Note that this is the full unified environment.
 
 .. code-block:: console
 
-   module use /glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.4.1/envs/unified-env
+   module use /glade/work/epicufsrt/contrib/spack-stack/cheyenne/spack-stack-1.5.0/envs/unified-env/install/modulefiles/Core
    module load stack-gcc/10.1.0
    module load stack-openmpi/4.1.1
    module load stack-python/3.9.12
@@ -688,33 +688,16 @@ For ``spack-stack-1.5.0`` with Intel on the JCSDA ROMEX cluster (``c6i.32xlarge`
 Amazon Web Services Red Hat 8
 -----------------------------
 
-Use a c6i.4xlarge instance or similar with AMI "skylab-5.0.0-redhat8" (ami-02324faac94a9cac9 in region us-east-1, ami-038d9beca351f9005 in region us-east-2).
+Use a c6i.4xlarge instance or larger if running out of memory with AMI "skylab-6.0.0-redhat8" (ami-MISSING in region us-east-1, ami-01fcf5d75ced5a046 in region us-east-2).
 
-For ``spack-stack-1.4.0``, run:
+For ``spack-stack-1.5.0``, run:
 
 .. code-block:: console
 
    ulimit -s unlimited
    scl enable gcc-toolset-11 bash
-   module use /home/ec2-user/spack-stack/spack-stack-1.4.0/envs/unified-env/install/modulefiles/Core
+   module use /home/ec2-user/spack-stack/spack-stack-1.5.0/envs/unified-env/install/modulefiles/Core
    module load stack-gcc/11.2.1
-   module load stack-openmpi/4.1.5
-   module load stack-python/3.10.8
-   module available
-
------------------------------
-Amazon Web Services Ubuntu 20
------------------------------
-
-Use a c6i.4xlarge instance or similar with AMI "skylab-5.0.0-ubuntu20" (ami-09a8c9d3775feafcf in region us-east-1, ami-03e47cdb4ced34d7e in region us-east-2).
-
-For ``spack-stack-1.4.0``, run:
-
-.. code-block:: console
-
-   ulimit -s unlimited
-   module use /home/ubuntu/spack-stack/spack-stack-1.4.0/envs/unified-env/install/modulefiles/Core
-   module load stack-gcc/10.3.0
    module load stack-openmpi/4.1.5
    module load stack-python/3.10.8
    module available

--- a/doc/source/PreConfiguredSites.rst
+++ b/doc/source/PreConfiguredSites.rst
@@ -311,7 +311,7 @@ The following is required for building new spack environments and for using spac
    module purge
    export LMOD_TMOD_FIND_FIRST=yes
    module use /glade/work/jedipara/cheyenne/spack-stack/modulefiles/misc
-   module load miniconda/3.9.12
+   module load python/3.7.9
    module load ecflow/5.8.4
    module load mysql/8.0.31
 

--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -24,9 +24,9 @@ copyright = '2023 '
 author = 'Dominikus Heinzeller, Alexander Richert, Cameron Book'
 
 # The short X.Y version
-version = '1.4'
+version = '1.5'
 # The full version, including alpha/beta/rc tags
-release = '1.4.1'
+release = '1.5.0'
 
 numfig = True
 

